californium-core/src/main/java/org/eclipse/californium/core/network/RandomTokenGenerator.java

```diff
@@ -56,16 +58,49 @@ public RandomTokenGenerator(final NetworkConfig networkConfig) {
 	}
 
 	@Override
-	public Token createToken(boolean longTermScope) {
+	public Token createToken(Scope scope) {
 		byte[] token = new byte[tokenSize];
 		rng.nextBytes(token);
-		if (longTermScope) {
+		switch (scope) {
+		case LONG_TERM:
 			// set bit 0 to 1
 			token[0] |= 0x1;
-		} else {
-			// set bit 0 to 0
-			token[0] &= 0xfe;
+			break;
+		case SHORT_TERM:
+			// set bit 1-0 to b10
+			token[0] &= 0xfc;
+			token[0] |= 0x2;
+			break;
+		case SHORT_TERM_CLIENT_LOCAL:
+			// set bit 1-0 to b00
+			token[0] &= 0xfc;
+			break;
 		}
 		return Token.fromProvider(token);
 	}
+
+	@Override
+	public Scope getScope(Token token) {
+		if (token.length() != tokenSize) {
+			return Scope.SHORT_TERM_CLIENT_LOCAL;
+		}
+		int scope = token.getBytes()[0] & 0x3;
+		switch (scope) {
+		case 0:
+			return Scope.SHORT_TERM_CLIENT_LOCAL;
+		case 2:
+			return Scope.SHORT_TERM;
+		}
+		return Scope.LONG_TERM;
```

Comment:
why not using the binary notation in the switch ?

Reply:
OK